deployment/7_upgrade_ens.js

```diff
@@ -0,0 +1,31 @@
+const TestENSRegistry = require('../build/TestENSRegistry');
+const TestReverseRegistrar = require('../build/TestReverseRegistrar');
+const ENSResolver = require('../build/ENSResolver');
+const ENSManager = require('../build/ENSManager');
+
+const deploy = async (network) => {
+    const config = configurator.config;
+    const domain = config.ENS.domain;
+    const deploymentAccount = await deployer.signer.getAddress();
+
+	// ENS domain
+	const parts = domain.split('.');
+	const extension = parts[1];
+	const domainName = parts[0];
+
+    // Instantiate the ENS Registry
+    const ensRegistry = await deployer.wrapDeployedContract(TestENSRegistry, config.ENS.ensRegistry);
+
+	// Create the 'argentx.xyz' wallet ENS namespace
+	const setSubnodeOwnerArgent = await ensRegistry.contract.setSubnodeOwner(utils.namehash(extension), utils.sha3(domainName), deploymentAccount);
+	await ensRegistry.verboseWaitForTransaction(setSubnodeOwnerArgent, `Setting Subnode Owner for ${domainName}.${extension}`);
+
+    const ensResolver = await deployer.deploy(ENSResolver);
+    const ensManager = await deployer.deploy(ENSManager, {}, domain, walletNode, config.ENS.ensRegistry, ensResolver.contractAddress);
+
+    const ensReverse = await deployer.deploy(TestReverseRegistrar, {}, ensRegistry.contractAddress, ensResolver.contractAddress);
+}
```

Comment:
Finally we need to upload the ABI and the address of the new ENSManager to AWS:
```
    configurator.updateInfrastructureAddresses({
        ENSManager: ENSManagerWrapper.contractAddress
    });
    await configurator.save();

    await Promise.all([
        abiUploader.upload(ENSManagerWrapper, "contracts")
    ]);
```

Reply:
What is the abi actually used for on AWS?

Reply:
The ABIs are exposed via a public end-point and some of them are used by the backend (e.g. the WalletFactory) or the clients (the modules, as part of the build process).